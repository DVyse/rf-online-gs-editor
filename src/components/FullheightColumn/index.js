/**
 *
 * FullheightColumn
 *
 */

import { AutoSizer } from 'react-virtualized';
import { Grid } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const columnStyle = { display: 'flex', flexDirection: 'column' };

class FullheightColumn extends React.PureComponent {
  render() {
    const { children, ...props } = this.props;

    return (
      <Grid.Column style={columnStyle} {...props}>
        {children}
      </Grid.Column>
    );
  }
}

FullheightColumn.propTypes = {
  children: PropTypes.node.isRequired,
};

export default FullheightColumn;

export function FullheightAutoSizer({ children }) {
  return (
    <AutoSizer>
      {({ height, width }) => (
        <FullheightAutoSizerChild height={height} width={width}>
          {children}
        </FullheightAutoSizerChild>
      )}
    </AutoSizer>
  );
}

FullheightAutoSizer.propTypes = {
  children: PropTypes.node.isRequired,
};

// styled

export const FullheightThis = styled.div`
  display: flex;
  flex: 1 100%;
  padding-top: ${({ paddingTop }) => paddingTop || 15}px;
  padding-bottom: ${({ paddingBottom }) => paddingBottom || 0}px;
`;

export const FullheightAutoSizerChild = styled.div`
  ${({ width, height }) => `
    width: ${width}px;
    height: ${height}px;
  `};

  overflow: hidden;
  overflow-y: scroll;
  padding-right: 15px;
`;
