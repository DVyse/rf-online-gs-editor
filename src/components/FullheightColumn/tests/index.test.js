import 'jest-styled-components';
import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import FullheightColumn, {
  FullheightThis,
  FullheightAutoSizer,
  FullheightAutoSizerChild,
} from '../index';

describe('<FullheightColumn />', () => {
  // test styled
  it('should exist FullheightThis', () => {
    const rendered = renderer.create(<FullheightThis />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
  it('should render FullheightThis with custom props', () => {
    const rendered = renderer
      .create(<FullheightThis paddingTop={10} paddingBottom={99} />)
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });
  it('should exist FullheightAutoSizerChild', () => {
    const rendered = renderer.create(<FullheightAutoSizerChild />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
  it('should render FullheightAutoSizerChild with custom props', () => {
    const rendered = renderer
      .create(<FullheightAutoSizerChild width={10} height={99} />)
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });
  it('should exist FullheightAutoSizer', () => {
    const wrapper = shallow(
      <FullheightAutoSizer>
        <div />
      </FullheightAutoSizer>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should exist FullheightColumn', () => {
    const wrapper = shallow(
      <FullheightColumn>
        <div />
      </FullheightColumn>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
