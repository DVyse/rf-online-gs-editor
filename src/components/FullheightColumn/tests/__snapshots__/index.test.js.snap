// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullheightColumn /> should exist FullheightAutoSizer 1`] = `
<AutoSizer
  disableHeight={false}
  disableWidth={false}
  onResize={[Function]}
  style={Object {}}
>
  <Component />
</AutoSizer>
`;

exports[`<FullheightColumn /> should exist FullheightAutoSizerChild 1`] = `
.c0 {
  width: undefinedpx;
  height: undefinedpx;
  overflow: hidden;
  overflow-y: scroll;
  padding-right: 15px;
}

<div
  className="c0"
/>
`;

exports[`<FullheightColumn /> should exist FullheightColumn 1`] = `
<GridColumn
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div />
</GridColumn>
`;

exports[`<FullheightColumn /> should exist FullheightThis 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 100%;
  -ms-flex: 1 100%;
  flex: 1 100%;
  padding-top: 15px;
  padding-bottom: 0px;
}

<div
  className="c0"
/>
`;

exports[`<FullheightColumn /> should render FullheightAutoSizerChild with custom props 1`] = `
.c0 {
  width: 10px;
  height: 99px;
  overflow: hidden;
  overflow-y: scroll;
  padding-right: 15px;
}

<div
  className="c0"
  height={99}
  width={10}
/>
`;

exports[`<FullheightColumn /> should render FullheightThis with custom props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 100%;
  -ms-flex: 1 100%;
  flex: 1 100%;
  padding-top: 10px;
  padding-bottom: 99px;
}

<div
  className="c0"
/>
`;
